### ISSUE TITLE
Socket error on client - but still receives MQTT messages.....

### ISSUE THREAD

**Issue by 877dev** [eclipse/mosquitto#2080]
I noticed this error in my Docker logs (set to log all).

I have a Pi3b with a bme280 temperature sensor attached to i2c bus.
A Python script is reading the sensor and publishing the three values to my mqtt broker.

The broker is receiving the three messages, but give a "socket error".

Any reason for this, and can I fix it? Or just ignore...
Mosquitto 1.16.12 and 1.16.13 give the same..

Log results:

```
2021-02-08T19:18:02: New client connected from 192.168.1.100 as skr_mini_temp (p2, c1, k60).
2021-02-08T19:18:02: Sending CONNACK to skr_mini_temp (0, 0)
2021-02-08T19:18:02: Received PUBLISH from skr_mini_temp (d0, q0, r0, m0, 'octoPrint/bme280/temperature', ... (12 bytes))
2021-02-08T19:18:02: Received PUBLISH from skr_mini_temp (d0, q0, r0, m0, 'octoPrint/bme280/humidity', ... (13 bytes))
2021-02-08T19:18:02: Received PUBLISH from skr_mini_temp (d0, q0, r0, m0, 'octoPrint/bme280/pressure', ... (12 bytes))
2021-02-08T19:18:02: Socket error on client skr_mini_temp, disconnecting.
```

And the Python code:
```
#!/usr/bin/python

import smbus2
import bme280
import paho.mqtt.client as mqtt #import the client1

# Setup bme280 sensor
port = 1
address = 0x76
bus = smbus2.SMBus(port)

#Setup mqtt broker
broker_address="192.168.1.99"


#Read the sensor
# the sample method will take a single reading and return a compensated_reading object
calibration_params = bme280.load_calibration_params(bus, address)
data = bme280.sample(bus, address, calibration_params)



# publish temperature to mqtt
client = mqtt.Client("skr_mini_temp") #create new instance
client.connect(broker_address) #connect to broker
client.publish("octoPrint/bme280/temperature",(data.temperature))	#publish temperature
client.publish("octoPrint/bme280/humidity",(data.humidity))		#publish humidity
client.publish("octoPrint/bme280/pressure",(data.pressure))		#publish pressure


```

**Comment by ralight** [comment]
You're right, that message usually just means "the client closed the connection without sending DISCONNECT". In your first case, it's possible the program ended before the connection was completed and the messages were sent.

Using `publish.multiple` is fine, but can I also suggest that you at least consider keeping a connection to the broker open. If you are using TLS encrypted connections then making a new connection is very CPU intensive, so if you reconnect every second or so then the TLS handshake will dominate your CPU usage. It's not a big deal if this is just for your own project, but it's at least worth me mentioning it.

**Comment by 877dev** [comment]
@ralight thank you for the advice. I agree with you that an established connection would be better, leaving more cpu available. 

As the Pi3b is only sometimes online (when 3d printing), would that be an issue?

I'm new to Python/Paho so not sure yet how to keep the connection open like that, and then disconnect gracefully when shutting the Pi down.

**Comment by ralight** [comment]
If it's just you and one/two/a few clients then I really wouldn't worry. If you're planning on turning it into something used by others, then it might be worth fixing.

I've experienced some MQTT applications that use TLS encryption, send a message every second and disconnect/reconnect every second, and that expect multiple clients to be connected. It's really wasteful.

You can see the difference in instructions used here: https://twitter.com/MosquittoMQTT/status/1293935424818352130

**Comment by 877dev** [comment]
No it's just for me at the moment, although I may post on my GitHub at some point as it's part of a larger project. 

Thanks for the link and advice 👍
